src/Functions/formatDateTime.cpp

```diff
@@ -941,6 +953,13 @@ class FunctionFormatDateTimeImpl : public IFunction
         auto * pos = begin;
         for (size_t i = 0; i < vec.size(); ++i)
         {
+            if (!const_time_zone_column && arguments.size() > 2)
+            {
+                if (!arguments[2].column.get()->getDataAt(i).toString().empty())
+                    time_zone = &DateLUT::instance(arguments[2].column.get()->getDataAt(i).toString());
+                else
+                    throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Provided time zone must be non-empty and be a valid time zone");
```

Comment:
Minor: I know this message was borrowed from extractTimeZoneFunctionArgument.cpp, but the error text `[...] and be a valid time zone` is a bit weird. We are only checking (here and in the other file) that the time zone is non-empty. We are not checking if it is "valid", e.g. "Fantasia/DisneyLand" is technically not rejected here (it is rejected in l. 959. Let's say in the error message only "... must be non-empty." (also in the other file, please)